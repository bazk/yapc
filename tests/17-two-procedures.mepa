     INPP
     AMEM 2
     DSVS R00
R01: ENPR 1 
     AMEM 1
     DSVS R02
R02: NADA 
     CRVL 1, -4
     CRVL 1, -4
     MULT
     ARMZ 1, 0
     CRVL 0, 0
     CRVL 0, 1
     SOMA
     CRVL 1, 0
     SOMA
     ARMZ 0, 0
     CRVL 0, 0
     IMPR
     DMEM 1
     RTPR 1, 1
R03: ENPR 1 
     AMEM 1
     DSVS R04
R05: ENPR 2 
     AMEM 1
     DSVS R06
R06: NADA 
     CRVL 0, 0
     CRVL 1, 0
     SOMA
     CRVL 2, -4
     SOMA
     ARMZ 2, 0
     CRVL 2, 0
     CHPR R01, 2
     CRVL 2, 0
     ARMZ 0, 0
     DMEM 1
     RTPR 2, 1
R04: NADA 
     CRVL 1, -4
     CRCT 1
     SOMA
     ARMZ 1, 0
     CRVL 1, 0
     CHPR R05, 1
     CRVL 0, 0
     CRVL 1, 0
     SOMA
     CHPR R01, 1
     DMEM 1
     RTPR 1, 1
R00: NADA 
     CRCT 1
     ARMZ 0, 0
     CRCT 3
     ARMZ 0, 1
     CRVL 0, 1
     CHPR R03, 0
     CRVL 0, 1
     CHPR R01, 0
     CRVL 0, 1
     IMPR
     CRVL 0, 0
     IMPR
     DMEM 2
     PARA
