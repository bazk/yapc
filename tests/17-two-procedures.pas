program exemplo74 (input, output);
var z, x: integer;
   procedure g(t:integer);
   var y : integer;
   begin
      y:=t*t; z:=z+x+y;
      write(z)
   end;

   procedure h (y : integer);
   var x : integer;
      procedure f(y : integer);
      var t : integer;
      begin
         t:=z+x+y; g(t);
         z:=t
      end;
   begin
      x:=y+1;
      f(x);
      g(z+x)
   end;
begin
   z:=1;
   x:=3;
   h(x);
   g(x);
   write(x,z)
end.
   

