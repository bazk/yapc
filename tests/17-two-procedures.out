85
181
193
3
193
